src/common/obj_bencher.cc

```diff
@@ -388,10 +388,18 @@ int ObjBencher::write_bench(int secondsToRun,
     }
     lock.Unlock();
     //create new contents and name on the heap, and fill them
-    newContents = new bufferlist();
     newName = generate_object_name(data.started);
-    snprintf(data.object_contents, data.object_size, "I'm the %16dth object!", data.started);
-    newContents->append(data.object_contents, data.object_size);
+    newContents = contents[slot];
+    if (!newContents) {
+        newContents = new bufferlist();
+        contents[slot] = newContents;
+        snprintf(data.object_contents, data.object_size, "I'm the %16dth object!", data.started);
+        newContents->append(data.object_contents, data.object_size);
+    } else {
+        // hacky way to invalidate internal crc cache
+        newContents->zero(0, 0);
+        snprintf(newContents->c_str(), data.object_size, "I'm the %16dth object!", data.started);
+    }
```

Comment:
What about having
<pre>
snprintf(newContents->c_str(), data.object_size, "I'm the %16dth object!", data.started);
</pre>
after the if () {} else {} ? And you could get rid of
<pre>
snprintf(data.object_contents, data.object_size, "I'm the %16dth object!", data.started);
</pre>
entirely. For a moment I wondered: why do we need to override **data.object_contents** ? But actually we don't, it's just used as a temporary buffer.